dist/tests/inverted-index-test.js

```diff
@@ -0,0 +1,138 @@
+    it('should be a class', function () {
+      expect(newIndex instanceof _invertedIndex2.default).toBe(true);
+      expect(newIndex instanceof Object).toBe(true);
+      expect(typeof newIndex === 'undefined' ? 'undefined' : _typeof(newIndex)).toBe('object');
+    });
+    it('Ensures a valid JSON array', function () {
+      var isValid = _invertedIndex2.default.isValidFile(_books2.default);
+      expect(isValid).toBe(true);
+    });
+
+    it('Ensure proper error response when file is invalid or empty', function () {
+      var isValid = _invertedIndex2.default.isValidFile(_invalidJson2.default);
+      expect(isValid).toBe(false);
+    });
+  });
+
+  describe('Tokenize String', function () {
+    it('Ensures it is available in class InvertedIndex', function () {
+      expect(_invertedIndex2.default.tokenize).toBeDefined();
+    });
+    it('Ensures it returns an array containing alphabets only', function () {
+      expect(_invertedIndex2.default.tokenize(demoWords)).not.toContain('%');
+    });
+
+    it('Ensures it returns an array containing alphabets only', function () {
+      expect(_invertedIndex2.default.tokenize(demoWords)).not.toContain('#');
+    });
+
+    it('Ensures it returns an array containing the correct number of words', function () {
+      expect(_invertedIndex2.default.tokenize(demoWords).length).toBe(4);
+    });
+  });
+
+  describe('Unique Words', function () {
+    it('Ensures It is available in class InvertedIndex', function () {
+      expect(_invertedIndex2.default.uniqueWords).toBeDefined();
+    });
+    it('Ensures it returns an array of words without duplicates', function () {
+      expect(_invertedIndex2.default.uniqueWords(demoWords).length).toBe(4);
+    });
+  });
+
+  describe('Read Book Data', function () {
+    it('Ensure createIndex is available in class InvertedIndex', function () {
+      expect(newIndex.createIndex).toBeDefined();
+    });
+    it('Ensures the JSON file is not empty', function () {
+      expect(newIndex.createIndex('emptyBook', _emptyBook2.default)).toBe('JSON file is Empty');
+      expect(newIndex.createIndex('books', _books2.default)).not.toBe('JSON file is Empty');
+    });
+  });
+
+  describe('Populate Index', function () {
+    it('Ensures Index is created Once JSON is read', function () {
+      expect(newIndex.index.books).toBeDefined();
+    });
+    it('Ensures words are mapped to their document location', function () {
+      expect(Object.keys(newIndex.index).length).toBe(1);
+      expect(Object.keys(newIndex.index.books).length).toBe(22);
+      expect(newIndex.index.books.inquiry).toEqual([0]);
+      expect(newIndex.index.books.this).toEqual([0, 1]);
+    });
+  });
+
+  describe('Get Index', function () {
+    it('Ensures index is correct', function () {
+      expect(newIndex.getIndex('books')).toBeDefined();
+      expect(Object.keys(newIndex.getIndex('books')).length).toBe(22);
+    });
+  });
+
+  describe('Search Index', function () {
```

Comment:
I like the way each of your describe function handles a unique aspect of your work to test. Altogether, it's a nice job!